src/main/java/org/ow2/proactive/catalog/service/CatalogObjectService.java

```diff
@@ -445,6 +445,13 @@ public CatalogObjectMetadata restore(String bucketName, String name, Long commit
         return resultKinds;
     }
 
+    /**
+     * @return all ordered content types for all objects in catalog
+     */
+    public TreeSet<String> getContentTypes() {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The return type of this method should be an interface such as "SortedSet" rather than the implementation "TreeSet". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1319)